placementApp/utils.py

```diff
@@ -0,0 +1,88 @@
+import xlwt
+import datetime
+
+#############################
+#   EXCEL SHEET GENERATION  #
+#############################
+
+
+def generate_xls(company):
+    wb = xlwt.Workbook(encoding="utf-8")  # Creating Workbook
+
+    positions = company.positions.all()
+    for position in positions:
+        wb = generate_sheet(wb, position)  # Creating separate sheets for each position
+
+    return wb
+
+
+def generate_sheet(wb, position):
+    ws = wb.add_sheet(position.title)  # Creating sheet
+
+    # Writing title of the sheet
+    sheet_title = (
+        "Applications for "
+        + position.title
+        + ", "
+        + position.company.name
+        + " "
+        + str(get_curent_year())
+    ).upper()
+    font_style = xlwt.XFStyle()
+    font_style.font.bold = True
+    font_style.font.height = 300
+    font_style.alignment.horz = 2
+    ws.write_merge(0, 0, 0, 5, sheet_title, font_style)
+
+    # Writing the column names
+    font_style = xlwt.XFStyle()
+    font_style.font.bold = True
+    font_style.alignment.horz = 2
+    columns = [
+        "SAP ID",
+        "Name of canditate",
+        "Email address",
+        "CGPA",
+        "Status",
+        "Submitted at",
+    ]
+    col_width = [
+        13 * 260,
+        26 * 260,
+        30 * 260,
+        7 * 260,
+        23 * 260,
+        18 * 260,
+    ]
+
+    row_num = 2
```

Comment:
The function is not able to generate excel sheet. It generates a file without any extension

Reply:
The downloaded file has .xls extension when I am testing it. I even specified ".xls" extension while naming the workbook in the views function

Reply:
In your views 
replace --> response["Content-Disposition"] = "attachment; filename=" + name_of_workbook
with --> response["Content-Disposition"] = "attachment; filename=" +name_of_workbook + ' " '
This will solve your problem.